llpc/CMakeLists.txt

```diff
@@ -89,32 +89,41 @@ if(ICD_BUILD_LLPC)
     # Export the LLVM build path so that it's available in XGL.
     set(XGL_LLVM_BUILD_PATH ${XGL_LLVM_BUILD_PATH} PARENT_SCOPE)
 
-    llvm_map_components_to_libnames(llvm_libs
-        lgc
-        AMDGPUAsmParser
-        AMDGPUCodeGen
-        AMDGPUDisassembler
-        AMDGPUInfo
-        Analysis
-        BinaryFormat
-        Core
-        Coroutines
-        LTO
-        ipo
-        BitReader
-        BitWriter
-        CodeGen
-        InstCombine
-        IRReader
-        Linker
-        MC
-        Passes
-        ScalarOpts
-        Support
-        Target
-        TransformUtils
-    )
-    target_link_libraries(llpc PUBLIC ${llvm_libs})
+    if (LLVM_LINK_LLVM_DYLIB)
```

Comment:
What about `BUILD_SHARED_LIBS`?

Reply:
In my (admittedly long time ago) experience, BUILD_SHARED_LIBS breaks too often. Have you tried using it?

I haven't tested it myself. My reading of AddLLVM.cmake is that BUILD_SHARED_LIBS doesn't seem to change how target_link_libraries is invoked.